recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,143 @@
+                "broker": [True, False],
+                "apps": [True, False],
+                "plugins": [True, False],
+                "with_cjson": [True, False],
+            }
+    default_options = {"shared": False,
+                        "with_tls": True,
+                        "clients": False,
+                        "broker": False,
+                        "apps": False,
+                        "plugins": False,  # TODO, there is some logic, just enabling plugin does not work, needs also something else
+                        "with_cjson": False ,
+    }
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.options.with_tls:
+            self.requires("openssl/1.1.1i")
+        if self.options.with_cjson:
+            self.requires("cjson/1.7.14")
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("Visual Studio build for any MT runtime is not supported")
+        if self.options.with_cjson: # see _configure_cmake for the reason
+            raise ConanInvalidConfiguration("Option with_cjson not yet supported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_THREADING"] = self.settings.os != "Windows"
+        self._cmake.definitions["WITH_PIC"] = self.settings.os != "Windows"
+        self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+        self._cmake.definitions["WITH_TLS"] = self.options.with_tls
+        self._cmake.definitions["WITH_CLIENTS"] = self.options.clients
+        self._cmake.definitions["WITH_BROKER"] = self.options.broker
+        self._cmake.definitions["WITH_APPS"] = self.options.apps
+        self._cmake.definitions["WITH_PLUGINS"] = self.options.plugins
+        self._cmake.definitions["DOCUMENTATION"] = False
+        if self.options.with_tls:
+            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+
+        # this does not work, and I do not know how to make cmake of mosquitto FIND_PACKAGE(cJSON) succeed
+        # if self.options.with_cjson:
+        #     self._cmake.definitions["cJSON_SEARCH_PATH"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+        #     self._cmake.definitions["cJSON_ROOT_DIR"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
Visual Studio not working with MT runtime is because you're not using a cmake wrapper.
The cmake wrapper sets the runtime: /MT, /MTd, /MD, /MDd...

Reply:
I am not sure what you mean, 
but I have spent now far more than 1 hour checking you comments and what you suggested, and nothing of this was something I could take over.

Please, do not wast my time like that. Do the following, 
Checkout the PR, make changes, test them on your machine, suggest changes when they work.

Reply:
Adding a recipe is not just a matter of fact of pushing it to this server and letting it bit rot.
When something is pushed to this repo, it must be maintained.

Also, I don't like your language. It's very disrespectful.

Reply:
Sorry for asking you to not suggest changes that break what is already working and cause others to waste a lot of time. Because I find it disrespectful to do that.

Reply:
It's working, because the recipe is not building what is not working.
I'm giving suggestions to not have this.

And github suggestions are just that: suggestions, no code. To give an idea.